node/status_node.go

```diff
@@ -179,14 +181,44 @@ func (n *StatusNode) setupRPCClient() (err error) {
 }
 
 func (n *StatusNode) discoveryEnabled() bool {
-	return n.config != nil && !n.config.NoDiscovery && n.config.ClusterConfig != nil
+	return n.config != nil && (!n.config.NoDiscovery || n.config.Rendezvous) && n.config.ClusterConfig != nil
+}
+
+func (n *StatusNode) startRendezvous() error {
+	if !n.config.Rendezvous {
+		return errors.New("rendezvous is not enabled")
+	}
+	if len(n.config.ClusterConfig.RendezvousNodes) == 0 {
+		return errors.New("rendezvous node must be provided if rendezvous discovery is enabled")
+	}
+	maddrs := make([]ma.Multiaddr, len(n.config.ClusterConfig.RendezvousNodes))
+	for i, addr := range n.config.ClusterConfig.RendezvousNodes {
+		var err error
+		maddrs[i], err = ma.NewMultiaddr(addr)
+		if err != nil {
+			return fmt.Errorf("failed to parse rendezvous node %s: %v", n.config.ClusterConfig.RendezvousNodes[0], err)
+		}
+	}
+	srv := n.gethNode.Server()
+	var err error
+	n.discovery, err = discovery.NewRendezvous(maddrs, srv.PrivateKey, srv.Self())
+	return err
 }
 
 func (n *StatusNode) startDiscovery() error {
-	n.discovery = peers.NewDiscV5(
-		n.gethNode.Server().PrivateKey,
-		n.config.ListenAddr,
-		parseNodesV5(n.config.ClusterConfig.BootNodes))
+	if !n.config.NoDiscovery && n.config.Rendezvous {
+		return errors.New("only one discovery can be used (will be allowed to use more in next change)")
+	}
+	if !n.config.NoDiscovery {
+		n.discovery = discovery.NewDiscV5(
+			n.gethNode.Server().PrivateKey,
+			n.config.ListenAddr,
+			parseNodesV5(n.config.ClusterConfig.BootNodes))
+	} else if n.config.Rendezvous {
+		if err := n.startRendezvous(); err != nil {
```

Comment:
I think we can extract Disocvery V5 structures into a separate method.

Reply:
i am not sure that i understood this comment. are you talking about different line?

Reply:
Yeah, I could put that in a different line :/ also this is a minor thing. This method is called `startDiscovery()` and it does `n.discovery = discovery.NewDiscV5()` or `n.startRendezvous()` which is a bit inconsistent, especially, that `n.startRendezvous()` just assigns rendezvous instance to `n.discovery` but not start it.

We could have something like
```go
if !n.config.NoDiscovery && n.config.Rendezvous {
  return errors.New(...)
}

n.discovery, err = s.getDiscovery(n.config)
if err != nil {
  return err
}

n.register = peers.NewRegister(n.discovery, n.config.RegisterTopics...)
...
```